umh-core/pkg/service/redpanda/redpanda.go

```diff
@@ -854,9 +849,8 @@ func (s *RedpandaService) IsLogsFine(logs []s6service.LogEntry, currentTime time
 			continue
 		}
 
-		for _, failure := range failures {
-			if strings.Contains(log.Content, failure) {
-
+		for _, failureDetector := range RedpandaFailures {
```

Comment:
can you explain how this is working? so that there is in redpanda_log_failures RedpandaFailures declared, with failure detectors, which all need to implement a IsFailure function

doesn't have to be long but some documentation here would be helpful